machine/quanta/quanta_common_rangeley/kernel/driver-support-broadcom-phy-bcm5461s.patch

```diff
@@ -0,0 +1,112 @@
+Add ethernet phy BCM5461S Support
+
+diff --git a/drivers/net/ethernet/intel/igb/e1000_82575.c b/drivers/net/ethernet/intel/igb/e1000_82575.c
+index 833f86f..c07d676 100644
+--- a/drivers/net/ethernet/intel/igb/e1000_82575.c
++++ b/drivers/net/ethernet/intel/igb/e1000_82575.c
+@@ -362,6 +362,9 @@ static s32 igb_get_invariants_82575(struct e1000_hw *hw)
+ 		phy->ops.get_cable_length   = igb_get_cable_length_82580;
+ 		phy->ops.get_phy_info       = igb_get_phy_info_82580;
+ 		break;
++	case BCM5461S_E_PHY_ID:
++		phy->type                   = e1000_phy_bcm5461s;
++		break;
+         case BCM54616_E_PHY_ID:
+ 		phy->type                   = e1000_phy_bcm54616;
+                 break;
+@@ -597,6 +600,10 @@ static s32 igb_phy_hw_reset_sgmii_82575(struct e1000_hw *hw)
+ 
+ 	ret_val = igb_phy_sw_reset(hw);
+ 
++	if (hw->phy.type == e1000_phy_bcm5461s) {
++		igb_phy_init_script_5461s(hw);
++	}
++
+ out:
+ 	return ret_val;
+ }
+@@ -1135,6 +1142,7 @@ static s32 igb_setup_copper_link_82575(struct e1000_hw *hw)
+ 	case e1000_phy_82580:
+ 		ret_val = igb_copper_link_setup_82580(hw);
+ 		break;
++	case e1000_phy_bcm5461s:
+ 	case e1000_phy_bcm54616:
+ 		ret_val = E1000_SUCCESS;
+ 		break;
+diff --git a/drivers/net/ethernet/intel/igb/e1000_defines.h b/drivers/net/ethernet/intel/igb/e1000_defines.h
+index 88be58d..0ab7463 100644
+--- a/drivers/net/ethernet/intel/igb/e1000_defines.h
++++ b/drivers/net/ethernet/intel/igb/e1000_defines.h
+@@ -694,6 +694,7 @@
+ #define I82580_I_PHY_ID      0x015403A0
+ #define I350_I_PHY_ID        0x015403B0
+ #define M88_VENDOR           0x0141
++#define BCM5461S_E_PHY_ID    0x002060C0
+ #define BCM54616_E_PHY_ID    0x03625D10
+ 
+ /* M88E1000 Specific Registers */
+diff --git a/drivers/net/ethernet/intel/igb/e1000_hw.h b/drivers/net/ethernet/intel/igb/e1000_hw.h
+index ab7c843..eda538d 100644
+--- a/drivers/net/ethernet/intel/igb/e1000_hw.h
++++ b/drivers/net/ethernet/intel/igb/e1000_hw.h
+@@ -121,6 +121,7 @@ enum e1000_phy_type {
+ 	e1000_phy_ife,
+ 	e1000_phy_82580,
+ 	e1000_phy_bcm54616,
++	e1000_phy_bcm5461s,
+ };
+ 
+ enum e1000_bus_type {
+diff --git a/drivers/net/ethernet/intel/igb/e1000_phy.c b/drivers/net/ethernet/intel/igb/e1000_phy.c
+index b17d7c2..7306ce3 100644
+--- a/drivers/net/ethernet/intel/igb/e1000_phy.c
++++ b/drivers/net/ethernet/intel/igb/e1000_phy.c
+@@ -2120,6 +2120,36 @@ s32 igb_phy_init_script_igp3(struct e1000_hw *hw)
+ }
+ 
+ /**
++ *  igb_phy_init_script_5461s - Inits the BCM5461S PHY
++ *  @hw: pointer to the HW structure
++ *
++ *  Initializes a Broadcom Gigabit PHY.
++ **/
++s32 igb_phy_init_script_5461s(struct e1000_hw *hw)
```

Comment:
BCM5461S LED polarity is not correct by default. This function sets the LED speed/activity to which Quanta defines 1000M -> Orange, 100M -> Green, 10M -> Off. Any better suggestions to apply this?